usersync/cookie.go

```diff
@@ -176,6 +176,9 @@ func (cookie *PBSCookie) GetId(bidderName openrtb_ext.BidderName) (id string, ex
 
 // SetCookieOnResponse is a shortcut for "ToHTTPCookie(); cookie.setDomain(domain); setCookie(w, cookie)"
 func (cookie *PBSCookie) SetCookieOnResponse(w http.ResponseWriter, setSiteCookie bool, cfg *config.HostCookie, ttl time.Duration) {
+	if !cfg.Refresh {
+		ttl = cookie.birthday.Add(ttl).Sub(time.Now())
```

Comment:
There is danger in using `time.Now()` to compute the ttl and another `time.Now()` call to compute the new expiration because the `time.Now()` values may be different and still trigger a refresh. I recommend a small refactor to use the exact "birthday" rather than converting to and from a ttl. Along the lines of....

```
func (cookie *PBSCookie) SetCookieOnResponse(w http.ResponseWriter, setSiteCookie bool, cfg *config.HostCookie, ttl time.Duration) {
	cookieExpiration := *cookie.birthday
	if cfg.Refresh {
		cookieExpiration = time.Now().Add(ttl)
	}
	httpCookie := cookie.ToHTTPCookie(cookieExpiration)
	...
	Expires: cookieExpiration,
	...
}
```
This would also require updating `ToHTTPCookie` to accept an expiration instead of a ttl. There are side effects, so perhaps it might be better to add a new `ToHTTPCookieWithExpiration` to differentiate it.